### ISSUE TITLE
KeyError: 'credential' when running  azure.azcollection.azure_rm_storageblob with azcollection 2.1.0

### ISSUE THREAD

**Issue by RikGr** [ansible/ansible_collections_azure#1375]
##### SUMMARY
When I try to run azcollection.azure_rm_storageblob in an Ansible playbook I receive this error since the latest version 2.1.0.
In version 2.0.0 it runs fine.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module > azure_rm_storageblob

##### ANSIBLE VERSION

```paste below
9.1.0
```

##### COLLECTION VERSION

```paste below
2.1.0
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

RedHat Linux 

##### STEPS TO REPRODUCE

```yaml
We run a playbook like this:
- name: Upload directories
  azure.azcollection.azure_rm_storageblob:
    state: present
    batch_upload_src: "{{ item.stat.path }}"
    batch_upload_dst: "{{ backup_directory }}{{ item.stat.path }}"
    subscription_id: "{{ subscriptionId }}"
    resource_group: "{{ resourceGroup }}"
    storage_account: "{{ storageAccount }}"
    container: backup
    auth_source: msi
  loop: "{{ directories.results }}"
  when: item.stat.exists
```

##### EXPECTED RESULTS

Upload of the files start

##### ACTUAL RESULTS

```paste below
 File \"/tmp/ansible_azure.azcollection.azure_rm_storageblob_payload_lvljfyyj/ansible_azure.azcollection.azure_rm_storageblob_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 439, in __init__\r\n  File \"/tmp/ansible_azure.azcollection.azure_rm_storageblob_payload_lvljfyyj/ansible_azure.azcollection.azure_rm_storageblob_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 1462, in __init__\r\nKeyError: 'credential'\r\n",

```

**Comment by surajupadhaya** [comment]
@RikGr 
I faced the same KeyError : 'credential' for module azure_rm_deployment 
Below steps could resolve the issue  : 
1. ansible-galaxy collection install azure.azcollection --force
2. pip3 install -r ~/.ansible/collections/ansible_collections/azure/azcollection/requirements-azure.txt
3. pip3 install ansible

**Comment by Fred-sun** [comment]
Fixed by #1393
